algorithms.py

```diff
@@ -572,6 +572,61 @@ def _search(self, values, query):
         return left
 
 
+def make_segment(left, right, node):
```

Comment:
These functions together feel like a class. Maybe a higher level interface like OverlapClounter or something would be good? It could have a method ``increment(left, right)`` which would do that, and hide away the details of how it's done.

Reply:
Yes this is much cleaner. Thanks!